numba/core/typing/npydecl.py

```diff
@@ -501,6 +501,17 @@ def typer(object, dtype=None):
                 dtype = _parse_dtype(dtype)
                 if dtype is None:
                     return
+            if isinstance(dtype, types.Record):
+                # a nested sequence should be interpreted as ndim=1 array,
+                # with the other dimension being part of the Record
+                if ndim == 1:
```

Comment:
I recommend you add a `stop_at_tuple=True` argument to `_parse_nested_sequence`, rather than forbidding 2d record arrays